include/cpp2util.h

```diff
@@ -2084,24 +2087,90 @@ struct args_t
     auto end()    const -> iterator    { return iterator{ argc, argv, argc }; }
     auto cbegin() const -> iterator    { return begin(); }
     auto cend()   const -> iterator    { return end(); }
-    auto size()   const -> std::size_t { return cpp2::unsafe_narrow<std::size_t>(argc); }
+    auto size()   const -> std::size_t { return cpp2::unsafe_narrow<std::size_t>(ssize()); }
     auto ssize()  const -> int         { return argc; }
 
     auto operator[](int i) const {
-        if (0 <= i && i < argc) { return std::string_view{ argv[i] }; }
-        else                    { return std::string_view{}; }
+        if (0 <= i && i < ssize())     { return std::string_view{ argv[i] }; }
+        else                           { return std::string_view{}; }
     }
 
     mutable int        argc = 0;        //  mutable for compatibility with frameworks that take 'int& argc'
     char**             argv = nullptr;
 };
 
-inline auto make_args(int argc, char** argv) -> args_t
+inline auto make_args(int argc, char** argv) -> args
 {
-    return args_t{argc, argv};
+    return args{argc, argv};
 }
 
 
+//-----------------------------------------------------------------------
+//
+//  range: a half-open range of [first, last)
+//
+//-----------------------------------------------------------------------
+//
+template<typename T>
+struct range
+{
+    range(
+        T const& f, 
+        T const& l, 
+        bool include_last = false
+    ) 
+        : first{ f }
+        , last{ l } 
+    { 
+        if (include_last) { 
+            ++last; 
+        }
+    }
```

Comment:
[Suggestion] How about something like (I am not great at choosing names):
```suggestion
    enum class type {
        ExcludeLast,
        IncludeLast
    };

    range(
        T const& f, 
        T const& l, 
        type t = type::ExcludeLast
    )
        : first{ f }
        , last{ l } 
    { 
        if (t = type::IncludeLast) { 
            ++last; 
        }
    }
```
The point here is that calls for closed intervals would change from `range(f, l, true)` to `range(f, l, type::IncludeLast)`, which expresses intent better.

This could even go even further to something of the form:
```
    template<range_type Type>
    range { /*...*/ };
```
including `if constexpr`.

Reply:
Thanks! This is in the generated code, but still I'll take a look at making it more readable. (I do use a fair number of `bool` flags in cppfront...)

Re `if constexpr`: Right now the only place that flag is used is in the constructor, it's never needed later. So I think there's no opportunity for `if constexpr` to make simpler code paths, there's already a single logic for all ranges?